c2rust-analyze/src/dataflow/mod.rs

```diff
@@ -81,6 +89,39 @@ impl DataflowConstraints {
                 )
             }
 
+            fn subset_except(
+                &mut self,
+                _a_ptr: PointerId,
+                a_val: &PermissionSet,
+                _b_ptr: PointerId,
+                b_val: &PermissionSet,
+                except: PermissionSet,
+            ) -> (PermissionSet, PermissionSet) {
+                let old_a = *a_val;
+                let old_b = *b_val;
+
+                // These should be `const`s, but that produces `error[E0015]: cannot call
+                // non-const operator in constants`.
+
+                // Permissions that should be propagated "down": if the superset (`b`)
+                // doesn't have it, then the subset (`a`) should have it removed.
+                #[allow(bad_style)]
+                let PROPAGATE_DOWN = PermissionSet::UNIQUE;
+                // Permissions that should be propagated "up": if the subset (`a`) has it,
+                // then the superset (`b`) should be given it.
+                #[allow(bad_style)]
+                let PROPAGATE_UP = PermissionSet::READ
+                    | PermissionSet::WRITE
+                    | PermissionSet::OFFSET_ADD
+                    | PermissionSet::OFFSET_SUB
+                    | PermissionSet::FREE;
```

Comment:
To avoid duplicating these constants, could you implement `fn subset` in terms of `subset_except` (with an empty `except` argument)?